lib/components/InputBox/BaseInputBox.tsx

```diff
@@ -40,17 +42,49 @@ const BaseInputBox: FC<BaseInputBoxProps> = ({
   ...inputProps
 }) => {
   const [inputID] = useState(inputProps?.id || uniqueid("pebbles__input__"));
+  const [isPassword, setIsPassword] = useState(false);
+  const [showPwd, setShowPwd] = useState(false);
+  const [inputType, setInputType] = useState<HTMLInputTypeAttribute>();
+
+  useEffect(() => {
+    inputProps?.type === "password" && setIsPassword(true);
+    inputProps.type && setInputType(inputProps.type);
+  }, []);
+
+  useEffect(() => {
+    if (isPassword) {
+      showPwd ? setInputType("text") : setInputType("password");
+    }
+  }, [showPwd, isPassword]);
```

Comment:
same here, you don't need this state variable, you can remove this useEffect and make the flow more linear.